src/pythonic/test_primes.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import itertools
+from main import Primes, Sieve
+
+
+def test_sieve_limit():
+    limit = 10000
+    with Sieve(limit) as s:
+        assert s.upper_bound() >= limit
+
+
+def test_upper_bound_exception():
```

Comment:
How about `test_checking_above_upper_bound_is_an_error`?
A panic only occurs if it's above upper bound (>= limit by small amount)